### PR TITLE
Alter components by commands

### PR DESCRIPTION
**Pull request by SaloEater**
JSON serializer does not support nested Polymorphic serializers, that's why I removed ComponentAction.
Part of a https://github.com/MineInAbyss/Looty/pull/15

### CODE REVIEW

**Comment by 0ffz** [comment]
I think tests are failing, please verify locally

**Comment by 0ffz** [comment]
Actually, it might be because some of my new commits, likely not your fault but will delay how early i can merge this most likely.

File: components/src/main/kotlin/com/mineinabyss/geary/ecs/conditions/EntityComponentConditions.kt [hunk 1]
```diff
@@ -15,11 +16,11 @@ import kotlinx.serialization.Serializable
 @SerialName("geary:entity")
 public class EntityComponentConditions(
     @SerialName("has")
-    public val components: Set<String> = emptySet(),
+    public val names: Set<String> = emptySet(),
```
Comment (0ffz) [inline components/src/main/kotlin/com/mineinabyss/geary/ecs/conditions/EntityComponentConditions.kt]: Use something more descriptive than `names`. I think `componentNames` works best for this, as well as all other similar conditions.

File: components/src/main/kotlin/com/mineinabyss/geary/ecs/actions/components/ComponentActions.kt [hunk 2]
```diff
@@ -32,11 +34,14 @@ public class AddComponentAction(
  */
 @Serializable
 @SerialName("geary:remove")
-public data class RemoveComponentAction(override val components: Set<String>) : ComponentAction() {
+public data class RemoveComponentAction(private val components: Set<String>) : GearyAction() {
+    @Transient
+    private val _components = Components(components)
```
Comment (0ffz) [inline components/src/main/kotlin/com/mineinabyss/geary/ecs/actions/components/ComponentActions.kt]: You don't typically don't use `_` for private fields in Java unless they're a backing field (e.x. a private mutable list with a public counterpart that returns an immutable list).

Remove in this case.

File: components/src/main/kotlin/com/mineinabyss/geary/ecs/helper/ComponentsHelper.kt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.mineinabyss.geary.ecs.helper
+
+import com.mineinabyss.geary.ecs.api.GearyComponent
+import com.mineinabyss.geary.ecs.serialization.Formats
+import kotlin.reflect.KClass
+
+public class Components (names: Collection<String>) {
```
Comment (0ffz) [inline components/src/main/kotlin/com/mineinabyss/geary/ecs/helper/ComponentsHelper.kt]: I think `Components` is a bit too broad of a name. Also no space between class and `(`. (I always spam `Ctrl + Alt + L` to do auto formatting, but you can also turn it on as an automatic step before every commit in intellij :p)

**Review by 0ffz — CHANGES_REQUESTED** [review]
I definitely prefer doing composition here, this is a welcome change :+1:. Ignore my comments on the tests failing, this is the dev branch so it doesn't even matter.

Mostly requesting some small style changes.

File: components/src/main/kotlin/com/mineinabyss/geary/ecs/helper/ComponentsHelper.kt (hunk 3 above)
Comment (SaloEater) [reply components/src/main/kotlin/com/mineinabyss/geary/ecs/helper/ComponentsHelper.kt]: I don't think we need a class here anymore until we have at least 2 different props inside

File: components/src/main/kotlin/com/mineinabyss/geary/ecs/helper/ComponentsHelper.kt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package com.mineinabyss.geary.ecs.helper
+
+import com.mineinabyss.geary.ecs.api.GearyComponent
+import com.mineinabyss.geary.ecs.serialization.Formats
+import kotlin.reflect.KClass
+
+public fun mapComponentNamesToClasses(names: Collection<String>): Collection<KClass<out GearyComponent>> =
```
Comment (0ffz) [inline components/src/main/kotlin/com/mineinabyss/geary/ecs/helper/ComponentsHelper.kt]: Small thing but an extension function would be best here I think (i.e. `Colleciton<String>.toComponentClasses()`).

Try using IntelliJ's function refactor menu (`Ctrl + F6`) to do this, it's a huuuuge time-saver.

**Review by 0ffz — CHANGES_REQUESTED** [review]

File: components/src/main/kotlin/com/mineinabyss/geary/ecs/helper/ComponentsHelper.kt (hunk 4 above)
Comment (SaloEater) [reply components/src/main/kotlin/com/mineinabyss/geary/ecs/helper/ComponentsHelper.kt]: ✔️

**Review by 0ffz — APPROVED** [review]

**Pull request merged**